src/pull.go

```diff
@@ -82,6 +88,72 @@ func (g *Commands) Pull(byId bool) error {
 	return g.playPullChanges(nonConflicts, g.opts.Exports, opMap)
 }
 
+func autoRenameClashes(g *Commands, clashes []*Change) error {
+	type rename struct {
+		name string
+		change *Change
+	}
+	clashesMap := map[string][]rename{}
+
+	for _, clash := range clashes {
+		group := clashesMap[clash.Path]
+		if clash.Src == nil { // Is that even possible? In what situation?
+			continue
+		}
+		clashesMap[clash.Path] = append(group, rename{"", clash})
+	}
+
+	for path, group := range clashesMap {
+		ext := filepath.Ext(path)
+		name := strings.TrimSuffix(path, ext)
+		n := 0
+
+		for i, r := range group[1:] {  // we can leave first item with original name
+			var newName string
+			for {
+				newName = fmt.Sprintf("%v_%d%v", name, n, ext)
+				n++
+
+				dupCheck, err := g.rem.FindByPath(newName)
+				if err != nil && err != ErrPathNotExists {
+					return err
+				}
+
+				if dupCheck == nil {
+					newName = filepath.Base(newName)
+					break
+				}
+			}
+
+			clashesMap[path][i+1].name = newName
+			g.log.Logln(fmt.Sprintf("%v %v -> %v", path, r.change.Src.Id, newName))
+		}
+	}
+
+	if g.opts.canPrompt() {
+		if !promptForChanges("Proceed with the changes [Y/N] ? ") {
+			return ErrClashesDetected
+		}
+	}
+
+	numErrors := 0
+	for path, group := range clashesMap {
+		for _, r := range group[1:] {
```

Comment:
If we just put the items into a list we won't have to worry about remembering that we are forgetting the first element. Please do that.